ci: build gp6 on ubuntu-22.04

gmusya opened the pull request.

gmusya committed:
ci: build gp6 on ubuntu-22.04

gmusya committed:
Install ubuntu dependencies

gmusya committed:
Use /home/gmusya/local as installation directory

gmusya committed:
Build arrow

gmusya committed:
Fix arrow branch

gmusya committed:
Arrow: fix cmake directory

gmusya committed:
Install grpc

gmusya committed:
grpc: submodule update

gmusya committed:
Build Tea

gmusya committed:
Add build directory

gmusya committed:
Check if grpc installs absl

gmusya committed:
Use absl from arrow

gmusya committed:
add cmake prefix path

gmusya committed:
Install gp6

gmusya committed:
Try to download dependencies

gmusya committed:
Fix url in arrow

gmusya committed:
Do not make 'build' directory twice

gmusya committed:
Fix

gmusya committed:
Build GP

gmusya committed:
Install llvm-17

gmusya committed:
llvm

gmusya committed:
install clang

gmusya committed:
use clang

gmusya committed:
remove clang and clang++

gmusya committed:
install clang-17

gmusya committed:
clang 17

gmusya committed:
run with sudo

gmusya committed:
which

gmusya committed:
set clang version for tea

gmusya committed:
Free space

gmusya committed:
clear cache

gmusya committed:
Try clang 18

gmusya committed:
fix clang version

gmusya committed:
libc

gmusya committed:
try different approach

gmusya committed:
install gcc-13

gmusya committed:
Use gp6 from pivotal gp-gpdb-dev

gmusya committed:
rename directory

gmusya committed:
Use open-gpdb gpdb

gmusya committed:
Use ygp

gmusya committed:
without mdblocales

gmusya committed:
Try to build without xml

gmusya committed:
Use gpdb from arenadata

gmusya committed:
switch to python2

gmusya committed:
force

gmusya committed:
Change dependencies

gmusya committed:
Add missing header

gmusya committed:
Try to add default constructor

gmusya committed:
Try to fix uninitialized warning

gmusya committed:
Add CMAKE_DL_LIBS

gmusya committed:
Remote teapot executables

gmusya committed:
Restart CI actions

andr-sokolov commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,135 @@
+          mkdir build
+          git clone https://github.com/apache/arrow.git
+          cd arrow
+          git checkout maint-15.0.2
```
```suggestion
          git clone https://github.com/apache/arrow.git -b maint-15.0.2
          cd arrow
```

gmusya committed:
arrow: clone branch

Co-authored-by: Andrey Sokolov <sokolov.andrey.yurevich@gmail.com>

andr-sokolov commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,134 @@
+        run: |
+          git clone https://github.com/grpc/grpc.git
+          cd grpc
+          git checkout v1.62.3
```
```suggestion
          git clone https://github.com/grpc/grpc.git -b v1.62.3
          cd grpc
```

andr-sokolov commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,134 @@
+
+          git clone https://github.com/arenadata/gpdb.git
+          cd gpdb
+          git checkout 6.29.1_arenadata68
```
```suggestion
          git clone https://github.com/arenadata/gpdb.git -b 6.29.1_arenadata68
          cd gpdb
```

andr-sokolov commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,134 @@
+          cd gpdb
+          git checkout 6.29.1_arenadata68
+          git submodule update --init
+          ./configure --with-perl --with-python --with-libxml --with-gssapi --prefix=$HOME/local/gpdb
```
what about adding the `--enable-cassert` flag?

gmusya committed:
clone one branch for gpdb and grpc

Co-authored-by: Andrey Sokolov <sokolov.andrey.yurevich@gmail.com>

andr-sokolov commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,132 @@
+            -DgRPC_BUILD_CSHARP_EXT=OFF -DgRPC_BUILD_GRPC_CSHARP_PLUGIN=OFF \
+            -DgRPC_BUILD_GRPC_NODE_PLUGIN=OFF -DgRPC_BUILD_GRPC_OBJECTIVE_C_PLUGIN=OFF \
+            -DgRPC_BUILD_GRPC_PHP_PLUGIN=OFF -DgRPC_BUILD_GRPC_PYTHON_PLUGIN=OFF \
+            -DgRPC_BUILD_GRPC_RUBY_PLUGIN=OFF ..
```
By default grpc installs its own ssl library to $HOME/local. It leads to link time problems, because we link two ssl libraries. I suggest this change to use libssl-dev 
```suggestion
            -DgRPC_BUILD_GRPC_RUBY_PLUGIN=OFF  -DgRPC_SSL_PROVIDER:STRING=package ..
```

gmusya committed:
grpc: set ssl_provider to package

Co-authored-by: Andrey Sokolov <sokolov.andrey.yurevich@gmail.com>

andr-sokolov commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,132 @@
+          cmake -GNinja -DCMAKE_BUILD_TYPE=RelWithDebInfo -DTEA_DEPS_PATH=$HOME/local \
+            -DGreenplum_ROOT=$HOME/local/gpdb -DCMAKE_PREFIX_PATH=$HOME/local \
+            -DCMAKE_C_COMPILER=gcc-13 -DCMAKE_CXX_COMPILER=g++-13 ..
+          ninja
```
```suggestion
          patch -p1 -d _deps/iceberg-cxx-src/ < ../vendor/iceberg-expose-manifests-filtering.patch
          patch -p1 -d _deps/iceberg-cxx-src/ < ../vendor/iceberg-ignore-partition-validation.patch
          ninja
```

gmusya replied:
Applied patches in vendor/CMakeLists.txt

https://github.com/lithium-tech/tea/pull/1/changes/9a251be9d0d0c534647beef1b96ea14c962a8bbe

andr-sokolov commented:
@gmusya, Is the `logger` argument redundant? I removed it to make the compilation successful

```diff
diff --git a/tea/gpext/tea_reader.cpp b/tea/gpext/tea_reader.cpp
index 080f7b9..b24a442 100644
--- a/tea/gpext/tea_reader.cpp
+++ b/tea/gpext/tea_reader.cpp
@@ -992,7 +992,7 @@ std::shared_ptr<tea::samovar::SingleQueueClient> SamovarMakePlan(TeaContextPtr t
         });
 
         iceberg::ice_tea::ScanMetadata all_meta =
-            iceberg::ValueSafe(iceberg::ice_tea::GetScanMetadata(*entries_stream, *table_metadata, logger));
+            iceberg::ValueSafe(iceberg::ice_tea::GetScanMetadata(*entries_stream, *table_metadata));
 
         ValidateAllMetadata(get::Config(tea_ctx), all_meta);
 
diff --git a/tea/metadata/access_iceberg.cpp b/tea/metadata/access_iceberg.cpp
index 13016cc..03c8a23 100644
--- a/tea/metadata/access_iceberg.cpp
+++ b/tea/metadata/access_iceberg.cpp
@@ -155,7 +155,7 @@ std::pair<iceberg::ice_tea::ScanMetadata, PlannerStats> FromIcebergWithLocation(
       entries_stream = std::make_shared<EmptyIcebergStream>();
     }
 
-    return iceberg::ice_tea::GetScanMetadata(*entries_stream, *table_metadata, logger);
+    return iceberg::ice_tea::GetScanMetadata(*entries_stream, *table_metadata);
   }();
 
   if (result.ok()) {
```

gmusya commented:
> @gmusya, Is the `logger` argument redundant? I removed it to make the compilation successful

While publishing patches and resolving conflicts, I lost an update to the iceberg-cxx. We do need logger in the latest version of iceberg-cxx. I fixed an iceberg version in commit https://github.com/lithium-tech/tea/pull/1/changes/9a251be9d0d0c534647beef1b96ea14c962a8bbe

gmusya committed:
Change patch order

gmusya committed:
fix typo

gmusya committed:
Check if iceberg-cxx thirft sources were broken

andr-sokolov commented on vendor/CMakeLists.txt:
```diff
@@ -29,8 +29,13 @@ FetchContent_Declare(
+  GIT_REPOSITORY ${GITHUB}/lithium-tech/iceberg-cxx.git
+  GIT_TAG 088cfcff6fcf9a539a280a2f41012ac61b6594d4
+  PATCH_COMMAND git reset --hard HEAD && git clean -df
+    && git apply ${CMAKE_CURRENT_SOURCE_DIR}/iceberg-empty.patch
```
Why do we need to clean iceberg-cxx repository?
```suggestion
  PATCH_COMMAND git apply ${CMAKE_CURRENT_SOURCE_DIR}/iceberg-empty.patch
```

gmusya replied:
If you don't clean the repository, running cmake again will try to apply the same patch again, and build will fail

gmusya replied:
But the `git clean -df` command is redundant (since we're not adding new files), so I'll remove it

gmusya replied:
Or I'll just apply patches to iceberg-cxx itself

andr-sokolov replied:
> Or I'll just apply patches to iceberg-cxx itself

I think it will be better

gmusya committed:
iceberg-cxx thrift version was not v0.19.0

gmusya committed:
Reduce diff

gmusya committed:
Add end of line to cmakelists.txt

gmusya committed:
Try to run tests in CI

gmusya committed:
Try to use cache

gmusya committed:
install redis-server

gmusya committed:
Fix strange redis-server behavior

andr-sokolov commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,192 @@
+          sudo apt-get update
+          sudo apt-get install -y software-properties-common
+          sudo add-apt-repository -y ppa:ubuntu-toolchain-r/test
+          sudo apt-get update
```
Do we really need `apt-get update` here? `add-apt-repository` synchronizes the package cache

andr-sokolov commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,192 @@
+      - name: Build Tea
+        run: |
+          cd build
+          cmake -GNinja -DCMAKE_BUILD_TYPE=RelWithDebInfo -DTEA_DEPS_PATH=$HOME/local \
```
Why is CMAKE_PREFIX_PATH not enough?

```suggestion
          cmake -GNinja -DCMAKE_BUILD_TYPE=RelWithDebInfo \
```

gmusya committed:
Bump iceberg

gmusya committed:
Review fix

gmusya committed:
Disable tests

gmusya committed:
Try to remove TEA_DEPS_PATH

gmusya committed:
Fix build line

gmusya committed:
Try to enable unit tests

gmusya committed:
Fix TEA_VERSION

gmusya committed:
Try to run e2e tests

gmusya committed:
install psutil via pip2

gmusya committed:
Revert "install psutil via pip2"

This reverts commit 76d8094eba773d10db528e6930c2635632d30e4b.

gmusya committed:
Add --with-pythonsrc-ext to gp configure

gmusya committed:
Change cache key

gmusya committed:
Make one successfull invocation to fill cache

gmusya committed:
Try to run tests in different job

gmusya committed:
Fix

gmusya committed:
Support profile

gmusya committed:
Use hive_metastore_server from iceberg-cxx

gmusya committed:
Build hms explicitly

gmusya committed:
Start redis

gmusya committed:
Some arrow patches

gmusya committed:
enable-cassert in gp

gmusya committed:
Revert "enable-cassert in gp"

This reverts commit 0af7e36dd0746737b7497be2d99aeb7afb49a57a.

gmusya committed:
Try to revert explicit default initialization

gmusya committed:
Remove duplicate

gmusya committed:
Add check-format

gmusya committed:
Try to remove uninitialized from muted warnings

andr-sokolov commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,324 @@
+            curl \
+            flex \
+            g++ \
+            gcc \
```
Do we really need to install g++ and gcc?  gcc-13 and g++-13 are installed later

andr-sokolov commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,324 @@
+      - name: Build and install gRPC
+        if: steps.cache-deps.outputs.cache-hit != 'true'
+        run: |
+          git clone https://github.com/grpc/grpc.git -b v1.62.3
```
I suggest to pass `--depth 1` to every `git clone` to reduce download time. For example, in the case of grpc we should download 500MB without  `--depth 1` and 20MB with it

andr-sokolov commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,324 @@
+          sudo add-apt-repository -y ppa:ubuntu-toolchain-r/test
+          sudo apt-get install -y gcc-13 g++-13
+          sudo systemctl stop redis-server
+          sudo systemctl disable redis-server
```
```suggestion
          sudo systemctl disable --now redis-server
```

andr-sokolov commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,324 @@
+            rsync \
+            zlib1g-dev
+          sudo apt-get update
+          sudo apt-get install -y software-properties-common
```
```suggestion
            zlib1g-dev \
            software-properties-common
```

andr-sokolov commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,324 @@
+            rsync \
+            zlib1g-dev
+          sudo apt-get update
+          sudo apt-get install -y software-properties-common
```
```suggestion
            zlib1g-dev \
            software-properties-common
```

andr-sokolov commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,324 @@
+          which gcc-13
+          which g++-13
+          sudo systemctl stop redis-server
+          sudo systemctl disable redis-server
```
```suggestion
          sudo systemctl disable --now redis-server
```

andr-sokolov commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,324 @@
+          sudo add-apt-repository -y ppa:ubuntu-toolchain-r/test
+          sudo apt-get install -y gcc-13 g++-13
+          which gcc-13
+          which g++-13
```
Why do we need this lines? What about removing them?

andr-sokolov commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,324 @@
+
+          source $HOME/local/gpdb/greenplum_path.sh
+          export MASTER_DATA_DIRECTORY=/gpdata/master/gpsne-1
+          NUM_SEGS=2 bash test/start-gp.sh $HOME/local/gpdb /gpdata
```
Why is this way better than using standard demo cluster (`make create-demo-cluster`)?

andr-sokolov commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,324 @@
+          cmake -GNinja -DCMAKE_BUILD_TYPE=RelWithDebInfo \
+            -DGreenplum_ROOT=$HOME/local/gpdb -DCMAKE_PREFIX_PATH=$HOME/local \
+            -DCMAKE_C_COMPILER=gcc-13 -DCMAKE_CXX_COMPILER=g++-13 \
+            -DTEA_USE_THREAD_SANITIZER=OFF -DTEA_VERSION=1.0 ..
```
may be TEA_VERSION should be 1.78.0?

gmusya committed:
Apply suggestions from code review

Co-authored-by: Andrey Sokolov <sokolov.andrey.yurevich@gmail.com>

gmusya committed:
Apply changes from review

gmusya committed:
Revert version

gmusya committed:
Add TODO

gmusya committed:
Fix comment in CI

andr-sokolov commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,317 @@
+          cmake -GNinja -DCMAKE_BUILD_TYPE=RelWithDebInfo \
+            -DGreenplum_ROOT=$HOME/local/gpdb -DCMAKE_PREFIX_PATH=$HOME/local \
+            -DCMAKE_C_COMPILER=gcc-13 -DCMAKE_CXX_COMPILER=g++-13 \
+            -DTEA_USE_THREAD_SANITIZER=OFF -DTEA_VERSION=1.79.0 ..
```
because https://github.com/lithium-tech/tea/pull/23 is merged
```suggestion
            -DTEA_USE_THREAD_SANITIZER=OFF ..
```

gmusya committed:
Update .github/workflows/ci.yml

Co-authored-by: Andrey Sokolov <sokolov.andrey.yurevich@gmail.com>

gmusya committed:
Do not use enable-cassert
